src/mavsdk/core/mavlink_parameter_subscription.cpp

```diff
@@ -0,0 +1,80 @@
+//
+// Created by consti10 on 21.06.22.
+//
+
+#include "mavlink_parameter_subscription.h"
+
+namespace mavsdk {
+
+template<class T>
+void MavlinkParameterSubscription::subscribe_param_changed(
+    const std::string& name, const ParamChangedCallback<T>& callback, const void* cookie)
+{
+    std::lock_guard<std::mutex> lock(_param_changed_subscriptions_mutex);
+    if (callback != nullptr) {
+        ParamChangedSubscription subscription{name, callback, cookie};
+        // This is just to let the upper level know of what probably is a bug, but we check again when actually calling the callback We also cannot assume here that the user called provide_param before subscribe_param_changed, so the only thing that makes sense is to log a warning, but then continue anyways.
+        /*std::lock_guard<std::mutex> lock2(_all_params_mutex);
+        if (_all_params.find(name) != _all_params.end()) {
+            const auto curr_value = _all_params.at(name);
+            if (!curr_value.template is_same_type_templated<T>()) {
+                LogDebug()
```

Comment:
I would say make this a `LogWarn` and enable it.

Reply:
This comes from before I refactored mavlink_parameter_subscription out. R.n. mavlink_parameter_subscription does not hold the parameter set, and therefore cannot do this check anymore.

Reply:
Oh ok. So the code can be deleted?

Reply:
Well, it would have been a nice (additional) check for the user, but as of r.n it is impractical - but that might change at some point.